sentry/src/main/java/io/sentry/util/EventSizeLimitingUtils.java

```diff
@@ -0,0 +1,177 @@
+package io.sentry.util;
+
+import io.sentry.Breadcrumb;
+import io.sentry.Hint;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.protocol.SentryException;
+import io.sentry.protocol.SentryStackFrame;
+import io.sentry.protocol.SentryStackTrace;
+import io.sentry.protocol.SentryThread;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Utility class that limits event size to 1MB by incrementally dropping fields when the event
+ * exceeds the limit.
+ */
+@ApiStatus.Internal
+public final class EventSizeLimitingUtils {
+
+  private static final long MAX_EVENT_SIZE_BYTES = 1024 * 1024;
+  private static final int FRAMES_PER_SIDE = 250;
```

Comment:
nit: How about calling this `MAX_FRAMES_PER_STACKTRACE` and then dividing it by `2` in the code below?
```suggestion
  private static final int MAX_FRAMES_PER_STACK = 500;
```